### DISCUSSION TITLE
Can we run the toolchain in Desktop PC which has NVIDIA RTX GPU?

### DISCUSSION THREAD

**Discussion by ghost** [FIRST-Tech-Challenge/fmltc#260] [Q&A] [answered]
Hi, 
Many of the students have gaming PC with good GPU in their homes. I was wondering if the FTC ML toolchain code can be run in a  PC/Laptop which has NVIDIA RTX GPU? This will help us not to buy time from Google cloud and run the toolchain for free without time limit. 
 
Thanks,
Tareq

**Comment by texasdiaz** [comment, marked as answer]
No, the ML Toolchain is not designed to run on a Desktop PC as-is, it is heavily designed for the Google Cloud infrastructure.

**Reply by ghost** [reply]
I downloaded the Datasets from FTC ML Toolchain. After extracting a dataset, the files have names like "train_dataset.record-00006-00052". Are they [TFRecord](https://www.tensorflow.org/tutorials/load_data/tfrecord) files? I was thinking to use these files and train them using Keras on a PC.

**Reply by lizlooney** [reply]
Yes they are tensorflow record files.

**Reply by ghost** [reply]
Thanks. 

When you train the model, are some layers frozen (i.e. parameters don't update) to preserve the data from transfer learning? Can you please provide some information which layers are trained? Thanks.
